File: src/main/kotlin/com/example/webservice/domains/users/models/dtos/RoleDto.java
package com.example.webservice.domains.users.models.dtos;

import com.example.webservice.domains.common.models.dtos.BaseDto;

import java.util.List;

public class RoleDto extends BaseDto {
    private String name;
    private List<PrivilegeDto> privileges;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<PrivilegeDto> getPrivileges() {
        return privileges;
    }

    public void setPrivileges(List<PrivilegeDto> privileges) {
        this.privileges = privileges;
    }
}

File: src/main/kotlin/com/example/webservice/domains/users/models/dtos/RoleDto.kt
package com.example.webservice.domains.users.models.dtos

import com.example.webservice.domains.common.models.dtos.BaseDto
import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

class RoleDto : BaseDto() {
    @NotBlank
    lateinit var name: String

    @NotNull
    var restricted: Boolean = true

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    var privileges: List<PrivilegeDto>? = null

    @NotNull
    @NotEmpty
    @JsonProperty(value = "privilege_ids", access = JsonProperty.Access.WRITE_ONLY)
    lateinit var privilegeIds: List<Long>


}
